### ISSUE TITLE
Warning dialog with timer after settings `"base_keymap": "None"`

### ISSUE THREAD

**Issue by feelamee** [zed-industries/zed#22421]
### Check for existing issues

- [X] Completed

### Problem:

After setting `"base_kemap"` to `"None"` all bindings (except custom ofc) will be disabled.
To return them you need open config in another editor and modify `"base_keymap"` value.

### Solution I suggest

Open dialog with timer and _confirm_ button after settings `"base_keymap": "None"`.
After timer will be expired `"base_keymap"` should be returned to previous value.
If _confirm_ was pressed before expiration, then `"None"` should remains.

There is nothing functional or else - just to improve user experience.
F.e. I saw such dialog in Windows when changing screen rotating.

Probably, I'm ready to contribute solution if someone like this and can set me on the right path to.

### Zed Version and System Specs

_isn't necessarily for this feature_

### If applicable, add mockups / screenshots to help present your vision of the feature

![image](https://github.com/user-attachments/assets/4d79231d-85e7-4689-b2dc-50a1567c4b5d)

Sorry for watermark, haha - I have no windows))

**Comment by 0x2CA** [comment]
I don't think this is feasible, how can the user confirm that the keymap is invalid? This is just to prevent the wrong display settings of the window system from causing the user to not see the popup window, the display can be restored after a timeout. How can the user be sure that the user has made an incorrect change to the keymap?

**Comment by feelamee** [comment]
> This is just to prevent the wrong display settings of the window system from causing the user to not see the popup window

I'm not sure that only for this. I think this behavior also useful when new display settings just uncomfortable/unusable, e.g. swapped horizontally and vertically. 

> How can the user be sure that the user has made an incorrect change to the keymap?

In my situation - I tried to use keyboard but nothing worked. Not only hotkeys but just typing too.

**Comment by notpeter** [comment]
I understand the desire for such a feature, but this is not something we are likely prioritize in the near term. 

Zed does not present "None" as an option in the `welcome: keymap selector` so this behavior can only be triggered by manually editing your own settings.json.  Note, if you trigger this mode, there are many ways to undo this action without externally editing your settings file:
- Edit->Undo, File->Save
- Delete that line from your settings and press File->Save.
- Double click the `"None"` value and type anything (either a valid keymap or anything) and then press save.
